### PR TITLE
mgr/cephadm: add support for proper osd daemon tracking

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jschmid@suse.de>

connected to #34835

### CODE REVIEW

**Comment by jschmid1** [comment]
make check failures are legit unit test failures. adapting..

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2056,6 +2072,21 @@ def get_osd_uuid_map(self, only_up=False):
     def apply_drivegroups(self, specs: List[DriveGroupSpec]):
         return [self._apply(spec) for spec in specs]
 
+    def get_osdspec_affinity(self, osd_id: str) -> str:
+        ret, out, err = self.mon_command({
+            'prefix': 'osd metadata',
+            'id': int(osd_id),
+            'format': 'json'
```
Comment (liewegas) [inline src/pybind/mgr/cephadm/module.py:2079]: the mgr already has all osd metadata in memory: ``self.get('osd_metadata')``.

**Comment by jschmid1** [comment]
test results: http://pulpito.ceph.com/jschmid-2020-04-30_14:45:11-rados-wip-jschmid1-testing-2020-04-30-1530-distro-basic-smithi/

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -2056,6 +2072,22 @@ def get_osd_uuid_map(self, only_up=False):
...
+        })
+        if ret != 0:
+            self.log.warning(f"Caught error on calling 'osd metadata {osd_id}' -> {err}")
+            return ''
+        try:
+            metadata = json.loads(out)
+        except json.decoder.JSONDecodeError:
+            self.log.error(f"Could not decode json -> {out}")
+            return ''
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py:2088]: Hm. what about raising instead of silently fail here? if we're getting an invalid json, something is wrong.

Comment (jschmid1) [reply src/pybind/mgr/cephadm/module.py:2088]: using `self.get('osd_metadata')` this won't be needed anymore.

**Comment by jschmid1** [comment]
tests after fixes and rebase: http://pulpito.ceph.com/jschmid-2020-05-05_06:55:50-rados-wip-jschmid1-testing-2020-05-04-1738-distro-basic-smithi/

**Pull request merged**